dojo/importers/default_importer.py

```diff
@@ -225,9 +238,48 @@ def process_findings(
             # all data is already saved on the finding, we only need to trigger post processing
 
-            # to avoid pushing a finding group multiple times, we push those outside of the loop
+            # We create a signature for the post processing task so we can decide to apply it async or sync
             push_to_jira = self.push_to_jira and (not self.findings_groups_enabled or not self.group_by)
-            finding_helper.post_process_finding_save(finding, dedupe_option=True, rules_option=True, product_grading_option=True, issue_updater_option=True, push_to_jira=push_to_jira)
+            post_processing_task_signature = finding_helper.post_process_finding_save_signature(
+                finding,
+                dedupe_option=True,
+                rules_option=True,
+                product_grading_option=False,
+                issue_updater_option=True,
+                push_to_jira=push_to_jira,
+            )
+
+            post_processing_task_signatures.append(post_processing_task_signature)
+
+            # Check if we should launch a chord (batch full or end of findings)
+            if we_want_async(async_user=self.user) and post_processing_task_signatures:
+                # Calculate current batch size: 2^batch_number, capped at max_batch_size
+                # We do this because post processing only starts after all tasks have been added to the chord
+                # So we start with small batches to minmize the delay
+                current_batch_size = min(2 ** current_batch_number, max_batch_size)
+
+                batch_full = len(post_processing_task_signatures) >= current_batch_size
+
+                if batch_full or is_final_finding:
+                    # Launch chord with current batch of signatures
+                    product = self.test.engagement.product
+                    system_settings = System_Settings.objects.get()
+                    if system_settings.enable_product_grade:
+                        calculate_grade_signature = utils.calculate_grade_signature(product)
+                        chord(post_processing_task_signatures)(calculate_grade_signature)
+                    elif post_processing_task_signatures:
+                        # If product grading is disabled, just run the post-processing tasks without the grade calculation callback
+                        group(post_processing_task_signatures).apply_async()
+
+                    logger.debug(f"Launched chord with {len(post_processing_task_signatures)} tasks (batch #{current_batch_number}, size: {len(post_processing_task_signatures)})")
+
+                    # Reset for next batch (only if not final)
+                    post_processing_task_signatures = []
+                    if not is_final_finding:
+                        current_batch_number += 1
+            else:
+                # Execute task immediately for synchronous processing
+                post_processing_task_signature()
```

Comment:
This block is written twice total. Can it be moved to the base importer as well?